### PR TITLE
Revert stopgaps related to beta server cert issue

### PR DESCRIPTION
**Pull request by sivaraam**
### Description (required)
The upstream issue with Commons beta server has been
[fixed now](https://phabricator.wikimedia.org/T243881#5861983). So, there's no point in stopgapping
the issue anymore. So, revert the related changes.

This reverts fa87eb566128c777dde3afc79ba2b68ce70efd55 and df426f7c427057fe97dc79186287260c02d3de25 which
correspond to PRs #3350 and #3349 respectively.

Related to #3320 and #3345.

### Tests performed (required)
Tested `betaDebug` on Pixel 3a (emulator) with API level 29.

### CODE REVIEW

**Comment by maskaravivek** [comment]
I would suggest not to revert the fix as this issue has happened multiple times in the past and has affected a lot of volunteers who are just getting started and are not aware that they can use the prod server for testing. Ref. #3320 and #2875. In both instances, we spent 1-2 weeks debugging the issue to make sure it is not an app-specific issue. 

Let us keep the fix as it simply bypasses SSL cert validation for beta servers. 

If the intention is to get a red flag when the beta server's cert expires, we can have a unit test or instrumented test for the same.

**Comment by sivaraam** [comment]
> I would suggest not to revert the fix as this issue has happened multiple times in the past and has affected a lot of volunteers who are just getting started and are not aware that they can use the prod server for testing. Ref. #3320 and #2875. In both instances, we spent 1-2 weeks debugging the issue to make sure it is not an app-specific issue.

I understand the emotion here. But from what I understand from the recent fix, I don't think we'll face this issues anymore. In both cases the issue was with an outdated OCSP staple in the certificate. According to the [SSL Labs report](https://www.ssllabs.com/ssltest/analyze.html?d=commons.wikimedia.beta.wmflabs.org) they've totally disabled OCSP stapling.

image

Not sure why. Have asked in the phab ticket about [whether this was intentional](https://phabricator.wikimedia.org/T243881#5862125).

> Let us keep the fix as it simply bypasses SSL cert validation for beta servers.

That's a [dangerous thing](https://www.troyhunt.com/heres-why-your-static-website-needs-https/) to do! It totally invalidates the security of the HTTPS connection to the beta servers. I do realise that we're doing this just for the beta version of the app. But I believe that it's *never good* to loosen the security. That would put the users in unpredictable and unforseeable risk. It's bad even if it might be just the app developers who use the beta version. So, I highly recommend against keeping this fix.

**Comment by misaochan** [comment]
I wonder if the best compromise would be to revert the stopgap fix, but make it clearer to new volunteers that they CAN use the prod server if they upload a Commons-worthy photo? IMO the majority of bug or good first issues (which they would be working on anyway) would not necessarily require the upload of more than a few photos for testing. Besides, it could be helpful for educating students on what the app does and the user's viewpoint. :)

Maybe we could mention this in the GSoC pinned issue?

**Comment by sivaraam** [comment]
> I wonder if the best compromise would be to revert the stopgap fix, but make it clearer to new volunteers that they CAN use the prod server if they upload a Commons-worthy photo?

That sounds like a good solution to me 👍 

> And there are so many things that would be welcome in Commons, that one could almost certainly take a photo from anywhere, even in their house or next to it. It wouldn't win any competitions, but it would still be valued. 

💯

**Comment by maskaravivek** [comment]
@sivaraam Can you check the travis build. Once that is sorted, we can merge the PR. :)

**Comment by sivaraam** [comment]
> @sivaraam Can you check the travis build. Once that is sorted, we can merge the PR. :)

Sure, I'll look into it.

For the note, I verified the fix when I submitted the PR but due to an [unrelated issue](https://phabricator.wikimedia.org/T244951) I'm currently unable to log-in to the Commons beta site (both via the app and the web interface).

File: app/src/test/kotlin/fr/free/nrw/commons/TestCommonsApplication.kt [hunk 1]
```diff
@@ -1,6 +1,5 @@
 package fr.free.nrw.commons
```
Comment (sivaraam) [inline app/src/test/kotlin/fr/free/nrw/commons/TestCommonsApplication.kt]: Reverting the changes in this file to the way they were in https://github.com/commons-app/apps-android-commons/commit/fa87eb566128c777dde3afc79ba2b68ce70efd55 seems to fix the CI issue. I couldn't see why this is the case. Can anyone shed some light about this?

Comment (sivaraam) [reply app/src/test/kotlin/fr/free/nrw/commons/TestCommonsApplication.kt]: cc: @ashishkumar468

**Comment by sivaraam** [comment]
I've pushed the fix for the failing CI issue. It's been some time and I've not been able to get into exploring _why_ the fix works. So, I thought I'll explore the reason later and just pushed the fix. I think we can merge this now 🙂

**Pull request merged**